### ISSUE TITLE
Utility function to generate ignore patterns

### ISSUE THREAD

**Issue by ark120202** [typescript-eslint/typescript-eslint#997]
Now that 2.0.0 requires all linted files to be included in projects it became even more important to ensure that eslint processes only actual source files. Today it's done with a `.eslintignore` file, but it can be quite hard to manage for projects with complex TypeScript configurations. Soon it would be possible to define ignored patterns in eslint config - https://github.com/eslint/eslint/pull/12274, which means it can be generated from actual patterns used to create programs. That also may simplify transition from TSLint, which processed only files included in tsconfig by default.

The new API could look like:
```ts
const { createIgnorePatterns } = require('@typescript-eslint/parser');
// function createIgnorePatterns(project: string | string[], tsconfigRootDir?: string): string[];

const project = [
  './project1/tsconfig.json',
  './project2/tsconfig.json',
];

module.exports = {
  ignorePatterns: createIgnorePatterns(project),
  parserOptions: { project },
};
```

**Comment by JoshuaKGoldberg** [comment]
@bradzacher before I start working on this one, just making sure - is this still something you're 👍 on? I personally am.

**Comment by bradzacher** [comment]
Yeah sounds good. I guess exported from the `utils` package?

Eventually I'd want it in our "core" package we use for flat configs.

**Comment by JoshuaKGoldberg** [comment]
> from the `utils` package

Hmm, most users don't take a direct dependency on that. I think it'd be simpler to export from `parser`, no?

**Comment by bradzacher** [comment]
Most users don't, no, as it's not necessary for config files.
I'd probably just hold off on this until we build our flat config tooling then?
Then we can create a user-focused package with all our tooling bundled up and some utils to make things easier?

**Comment by JoshuaKGoldberg** [comment]
Yeah that makes sense. And the fact that 0 users have 👍d this issue or filed duplicates of it since September 2019 indicates to me that it's not a common request. Closing for the same reasons as #5908: there's no strong user demand to be met right now.
